client/allocrunner/network_manager_nonlinux.go

```diff
@@ -14,9 +14,21 @@ import (
 	"github.com/hashicorp/nomad/plugins/drivers"
 )
 
+// noopNetworkManager implements the drivers.DriverNetoworkManager interface to
+// provide a no-op manager for systems that don't support network isolation.
+type noopNetworkManager struct{}
+
+func (*noopNetworkManager) CreateNetwork(_ string, _ *drivers.NetworkCreateRequest) (*drivers.NetworkIsolationSpec, bool, error) {
+	return nil, false, nil
```

Comment:
From what I've seen returning a `nil` spec is safe because access is guarded, but is there a safer zero-value that we could return?

Reply:
I guess technically `NetIsolationModeHost` is the "default", which would could represent with a non-nil isolation spec. Not sure how the plumbing around that works currently - my guess is we have `nil` checks and then do the host mode behavior

Reply:
Now that I think about it, returning a non-nil value here would evade the guards and potentially cause more problems if there are other assumptions about what's returned, so I think I will keep it `nil` for now. Thanks!